Akka 2.6.16 released (#812)
akka/akka.io: blog/_posts/2021-08-19-akka-2.6.16-released.md

---
layout: post
title: Akka 2.6.16 Released - including Durable State persistence
author: Patrik Nordwall

> short: We are excited to announce the 2.6.16, which include a new Durable State feature.

Reviewer: ```suggestion
short: We are excited to announce Akka 2.6.16, which includes a new Durable State feature.
```

category: news
tags: [releases]
---

Dear hakkers,

We are excited to announce a new patch release of Akka 2.6. Noteworthy additions and improvements in this release:

 * Addition of Durable State persistence, see below.

>  * Exclude exiting members in Read/Write MajorityPlus, which is used by Sharding (#30327)[https://github.com/akka/akka/issues/30327]

Reviewer 2: ```suggestion
 * Exclude exiting members in Read/Write MajorityPlus, which is used by Sharding [#30327](https://github.com/akka/akka/issues/30327)
```

 * Gracefully terminating shard region unnecessarily delays further rebalancing [#30401](https://github.com/akka/akka/issues/30401)
 * Coalesce writes in TCP Streams [#30334](https://github.com/akka/akka/pull/30334)
 * Possibility to configure log levels for backoff restarts for Streams and Actors [#30445](https://github.com/akka/akka/issues/30445)

2.6.16 includes 44 closed issues. The complete list can be found on the [2.6.16 milestone](https://github.com/akka/akka/milestone/177?closed=1) on github.

## Durable State

> [Durable State](https://doc.akka.io/docs/akka/2.6/typed/durable-state/persistence.html) is a new persistence model that complements Event Sourced persistence. It's useful when only the latest state is of interest. The full state is stored and it overwrites previously stored state. In contrast, Event Sourcing stores all events (changes) leading up to the current state and recovers the state by replaying the events.

Reviewer 2: maybe also link `Event Sourced persistence`?

Author: done

The API of `DurableStateBehavior` should be familiar to you that have already used `EventSourcedBehavior` and it is typically used together with Cluster Sharding. See examples in the [documentation](https://doc.akka.io/docs/akka/2.6.16/typed/durable-state/persistence.html#example-and-core-api).

> There is support for [queries](https://doc.akka.io/docs/akka/2.6.16/durable-state/persistence-query.html) and [Projections](https://doc.akka.io/docs/akka-projection/current/durable-state.html) to retrive a Stream of the state of all (tagged) entities.

Reviewer 2: maybe link to `2.6` rather than `2.6.16` for these 2 links, too?

Author: done

Database specific implementations have so far been added to the [JDBC](https://doc.akka.io/docs/akka-persistence-jdbc/current/) and [Spanner](https://doc.akka.io/docs/akka-persistence-spanner/current/) plugins of Akka Persistence.

> The API will be marked "may change" in a few patch releases before we declare it final. Take it for a spin and let us know what you think!

Reviewer 2: ```suggestion
The API will be marked "may change" for a few patch releases before we declare it final. Take it for a spin and let us know what you think!
```
